src/renderer.ts

```diff
@@ -204,10 +213,24 @@ class Renderer extends EventEmitter<RendererEvents> {
     return [div, shadow]
   }
 
+  /** Wavesurfer itself calls this method. Do not call it manually. */
   setOptions(options: WaveSurferOptions) {
+    if (this.options.container !== options.container) {
+      const newParent = this.parentFromOptionsContainer(options.container)
+      this.parent.removeChild(this.container)
```

Comment:
You don’t need to remove it before appending. AppendChild will move the element.

Reply:
Done